### PR TITLE
[APM-CI] Hey APM Pipeline version 0

### PR DESCRIPTION
**Pull request by kuisathaverat**
Pipeline that runs the unit and stress test of the Hey APM project, then store the results of the tests

screenshot 2018-11-15 at 15 47 32

image

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 .idea/
 vendor/
 hey-apm
+build
+Gopkg.lock
```
Comment (jalvz) [inline .gitignore]: what is the reason you added this? i don't think the lock file should be ignored, is there to make sure we have consistent versions of dependencies.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  options {
+    timeout(time: 1, unit: 'HOURS') 
+    buildDiscarder(logRotator(numToKeepStr: '20', artifactNumToKeepStr: '20', daysToKeepStr: '30'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+  }
+  parameters {
+    string(name: 'GO_VERSION', defaultValue: "1.10.3", description: "Go version to use.")
```
Comment (jalvz) [inline Jenkinsfile]: this is fine, but we can also jump to 1.11.2

File: scripts/jenkins/run-test.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+if [ ! -d "$APM_SERVER_DIR" ] ; then
+  echo "you need to define APM_SERVER_DIR environment variable pointing to the APM server source code"
+  exit 1
+fi
+
+eval "$(gvm ${GO_VERSION})"
+echo "Installing hey-apm dependencies and running unit tests..."
+go get -v -u github.com/golang/dep/cmd/dep
+go get -v -u github.com/graphaelli/hey/requester
```
Comment (jalvz) [inline scripts/jenkins/run-test.sh]: you just need `go get -v -u github.com/golang/dep/cmd/dep` and `dep ensure -v` to install all the rest? or is that what you had problems in the past?

File: scripts/jenkins/unit-test.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+set -exuo pipefail
+
+export GOPATH=$WORKSPACE
+export PATH=$PATH:$GOPATH/bin
+eval "$(gvm ${GO_VERSION})"
+echo "Installing hey-apm dependencies and running unit tests..."
+go get -v -u github.com/golang/dep/cmd/dep
+go get -v -u github.com/graphaelli/hey/requester
```
Comment (jalvz) [inline scripts/jenkins/unit-test.sh]: same as before. `dep ensure -v` and remove all `go gets` except the first one

File: Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+  }
+  options {
+    timeout(time: 1, unit: 'HOURS') 
+    buildDiscarder(logRotator(numToKeepStr: '20', artifactNumToKeepStr: '20', daysToKeepStr: '30'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+  }
+  parameters {
```
Comment (jalvz) [inline Jenkinsfile]: out of curiosity, what is the difference between these `parameters` and https://github.com/elastic/hey-apm/pull/69/files#diff-7a8b8e4479c1b69215ddb619ce195f8fR23?

File: scripts/jenkins/test-on-ci [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+  agent none
+  environment {
+    BASE_DIR="src/github.com/elastic/hey-apm"
+    APM_SERVER_BASE_DIR = "src/github.com/elastic/apm-server"
+    JOB_GIT_CREDENTIALS = "f6c7695a-671e-4f4f-a331-acdce44ff9ba"
+    GO_VERSION = "${params.GO_VERSION}"
+    APM_SERVER_VERSION = "${params.APM_SERVER_VERSION}"
+  }
+  triggers {
+    cron('0 0 * * 1-5')
```
Comment (jalvz) [inline scripts/jenkins/test-on-ci]: can you run this a bit later, say 3am utc or something? the onld (so current) tests in infra run at a similar time, i think against the same elasticsearch cluster, right? so would be better that they don't interfere with each other.

File: .gitignore (hunk 1 above)
Comment (kuisathaverat) [reply .gitignore]: ok, I removed it.

File: Jenkinsfile (hunk 2 above)
Comment (kuisathaverat) [reply Jenkinsfile]: done

File: Jenkinsfile (hunk 5 above)
Comment (kuisathaverat) [reply Jenkinsfile]: The Jenkinsfile is used by a multibranch pipeline job so do not need to specify a branch to build, for each branch/PR/tag will exist a project to run it.

File: scripts/jenkins/run-test.sh (hunk 3 above)
Comment (kuisathaverat) [reply scripts/jenkins/run-test.sh]: IIRC y made C&P from somewhere, I will change it.

File: scripts/jenkins/unit-test.sh (hunk 4 above)
Comment (kuisathaverat) [reply scripts/jenkins/unit-test.sh]: ditto

File: scripts/jenkins/test-on-ci (hunk 6 above)
Comment (kuisathaverat) [reply scripts/jenkins/test-on-ci]: I have changed it to be executed on the first available time-slot starting at 3:00 AM UTC

**Pull request merged**